lesson16/main.js

```diff
@@ -65,133 +26,164 @@ async function fetchData(url) {
       displayInfo("no data");
+    } else {
+      return responseData;
     }
-    return responseData;
   } catch (error) {
     displayInfo(error);
   } finally {
     removeCircle();
-    backButton.classList.remove("hidden");
   }
 }
 
-async function fetchRenderData(inputNumber) {
-  const responseData = await fetchData(url);
-  if (responseData) {
-    renderData(responseData);
-    console.log(inputNumber); //show inputted number
-  }
+function renderCircle() {
+  const loadingCircle = document.createElement("img");
+  loadingCircle.src = "img/loading-circle.gif";
+  loadingCircle.alt = "ローディング画像";
+  loadingCircle.id = "loading-circle";
+  tabArea.appendChild(loadingCircle);
 }
 
-openButton.addEventListener("click", () => {
-  resetPrompt();
-  nameBox.value = "";
-  numberBox.value = "";
-  modal.classList.remove("hidden");
-  mask.classList.remove("hidden");
-  openButton.classList.add("hidden");
-  fetchButton.setAttribute("disabled", "true");
-  nameBox.focus();
-});
-
-fetchButton.addEventListener("click", () => {
-  const inputName = nameBox.value;
-  const inputNumber = numberBox.value;
-  fetchRenderData(inputName, inputNumber);
-  modal.classList.add("hidden");
-  mask.classList.add("hidden");
-});
-
-function closeModal() {
-  modal.classList.add("hidden");
-  mask.classList.add("hidden");
-  openButton.classList.remove("hidden");
+function renderStatus(response) {
+  errorMessage.id = "render-status";
+  errorMessage.textContent = `${response.status}:${response.statusText}`;
+  document.body.appendChild(errorMessage);
 }
 
-mask.addEventListener("click", () => {
-  closeModal();
-  resetValidation();
-});
-
-closeButton.addEventListener("click", () => {
-  closeModal();
-  resetValidation();
-});
-
-backButton.addEventListener("click", () => {
-  backButton.classList.add("hidden");
-  openButton.classList.remove("hidden");
-  if (errorMessage) {
-    errorMessage.remove();
-  }
-  while (ul.firstChild) {
-    ul.firstChild.remove();
-  }
-});
-
-let isValidateName = false;
-let isValidateNumber = false;
+function displayInfo(error) {
+  errorMessage.id = "display-info";
+  errorMessage.textContent = error;
+  document.body.appendChild(errorMessage);
+}
 
-function validatePattern(inputBox, validPattern, errorMessage) {
-  const isValue = checkInputValue(inputBox, validPattern, errorMessage);
-  if (inputBox === nameBox) {
-    isValidateName = isValue;
-  } else if (inputBox === numberBox) {
-    isValidateNumber = isValue;
-  }
-  if (isValue) {
-    resetPrompt();
-    checkEnableSubmit();
+const fragmentTabs = document.createDocumentFragment();
+const fragmentGenres = document.createDocumentFragment();
+const fragmentTitles = document.createDocumentFragment();
+const fragmentImages = document.createDocumentFragment();
+
+function renderData(dataSet) {
+  console.log(dataSet);
+  for (const data of dataSet) {
+    console.log(data);
+    createTab(data);
+    createArticles(data);
+    const img = createImage(data);

```

Comment:
IMO
名前が抽象的だと思うので、もう少し用途に合わせて良いと思いました
```suggestion
    const img = createThumbnail (data);
```

Reply:
Thank you sae-san :)
I changed it to appropriate name.